zerver/lib/stream_subscription.py

```diff
@@ -76,6 +79,12 @@ def get_stream_subscriptions_for_user(user_profile: UserProfile) -> QuerySet[Sub
     )
 
 
+def get_user_subscribed_streams(user_profile: UserProfile) -> QuerySet[Stream]:
+    return Stream.objects.filter(
+        recipient_id__in=get_stream_subscriptions_for_user(user_profile).values("recipient_id")
+    )
```

Comment:
I think this would do better using `get_subscribed_stream_recipient_ids_for_user`

Reply:
done